### DISCUSSION TITLE
复制评论地址功能, 有什么替代实现方案么

### DISCUSSION THREAD

**Discussion by XMB-7** [the1812/Bilibili-Evolved#3807] [提问 / Q&A] [answered]
暂时看这个功能似乎是实现不了了, 看issue区里有人说返回旧版也不行了, 请问可能有什么替代的实现方案么. 比如发现某个评论及评论的评论很有价值, 如何才能进行标记收藏. 
正常复制内容, 期待日后再行搜索定位不可行, 因为多数时候某条评论需要不停下拉刷新出来, 才能被搜索到.

**Comment by the1812** [comment]
直接截图或者复制内容保存下来?

**Reply by XMB-7** [reply]
有时候某个楼的评论能有好几页，然后有价值的信息夹杂其中。此时无论是截图还是复制内容都是很麻烦的，不像之前直接复制评论地址就可以了，这样可以闲暇时候再回来整理...

**Comment by Tinhone** [comment]
好像改cookie可以返回旧版评论区

**Reply by XMB-7** [reply]
我看issue区有人说改回去也没用了，旧版也跟着改了。

**Reply by Tinhone** [reply]
怪哦，我这边旧版还用的正常

**Comment by Ethkuil** [comment]
从 HTML 角度上看是无法将不同评论区分开的，没有 id。要实现这个功能只能看 B站 有无合适的 API 了。

**Reply by Ethkuil** [reply]
[B站API - 评论区明细](https://github.com/SocialSisterYi/bilibili-API-collect/blob/master/comment/list.md)

只要能弄到 根回复的id 就可以了。亲测有效。但根回复的id要怎么获取我就不清楚了。

**Reply by XMB-7** [reply]
这个功能之前是好使的，开发者知道怎么获取，但说是B站页面改了就不行了，已经好几个月了，感觉可能是已经不让获取了...

**Comment by Geeyun-JY3** [comment]
`复制评论链接` 组件用不了吗？

而且要获取评论的 `data-id` 开 F12 不就完了么。
拿到 `data-id` 再按格式填进去，得到的链接肯定是能跳转到对应评论的。

另外我这边 [Bilibili 翻页评论区](https://greasyfork.org/scripts/37970) 也一直能用。

**Reply by the1812** [reply]
新版视频页拿不到 id 了

**Reply by Geeyun-JY3** [reply]
至少一直以来的那种格式的评论链接在新版视频页还能用（甚至还会[高亮出目标评论](https://github.com/the1812/Bilibili-Evolved/issues/1264#issuecomment-1266718605)），说明还是能通过链接定位到评论的。

去新版视频页看了一下，好像确实是藏得深了不少，开 F12 找还是能找到组成评论链接的 `data-id`，就是要花上更多时间，确实是麻烦了不少。

**Reply by XMB-7** [reply]
请问意思是说，之前复制的评论地址仍然有效，并且如今跳转还给出了高亮特效，所以说明评论id其实还在，但就是不暴露给外界了是么?

**Reply by Geeyun-JY3** [reply]
> 请问意思是说，之前复制的评论地址仍然有效，并且如今跳转还给出了高亮特效，所以说明评论id其实还在，但就是不暴露给外界了是么?

有从点赞和回复评论跳转的需求在就不会让之前的评论链接失效的。

反正我今天在无痕窗口测试过新版播放页，开 F12 还是能找到评论的 id，就是藏得深了点，不过好像也只能找到普通楼层的，楼中楼的似乎找不到了。

***

新版播放页改得属实恶臭，改完瀑布流还能变本加厉改得体验这么差，评论按时间排序还要登录。

**Reply by Tinhone** [reply]
> > 请问意思是说，之前复制的评论地址仍然有效，并且如今跳转还给出了高亮特效，所以说明评论id其实还在，但就是不暴露给外界了是么?
> 
> 有从点赞和回复评论跳转的需求在就不会让之前的评论链接失效的。
> 
> 反正我今天在无痕窗口测试过新版播放页，开 F12 还是能找到评论的 id，就是藏得深了点，不过好像也只能找到普通楼层的，楼中楼的似乎找不到了。
> 
> 新版播放页改得属实恶臭，改完瀑布流还能变本加厉改得体验这么差，评论按时间排序还要登录。

https://github.com/SocialSisterYi/bilibili-API-collect/blob/master/comment/list.md#获取指定评论的回复
通过监听用户翻页操作，貌似可以实现楼中楼链接的复制

<del>_有时候在深夜用新版播放页时还能看到在发行版里加注释的奇观_</del>

**Comment by the1812** [comment, marked as answer]
我找到 ID 了, a53da386 已实现

**Comment by XMB-7** [comment]
@the1812 麻烦看一下, 评论的下拉菜单里没有出现复制评论链接的选项, 相关代码可能被注释了, 感谢. 当前使用版本v2.5.2-17-g89d37a1f2
![image](https://user-images.githubusercontent.com/65625632/202965558-ec349bf6-2d93-486e-980f-4bee852e2e3e.png)

**Reply by the1812** [reply]
正式版还没有的

**Reply by Kal-tsit63** [reply]
打扰了，2.6版本还没有更新上吗？

**Reply by the1812** [reply]
还没有, 新版评论区还有些 bug
